src/http.cc

```diff
@@ -1933,6 +2071,49 @@ HttpStateData::httpBuildRequestHeader(HttpRequest * request,
     strConnection.clean();
 }
 
+void
+HttpStateData::makeUpgradeHeaders(HttpHeader &hdr_out)
+{
+    // We need virgin request received by squid. This is stored to
+    // AccessLogEntry::request member.
+    const HttpHeader &hdr_in = fwd->al->request ? fwd->al->request->header :request->header;
+
+    if (!hdr_in.has(Http::HdrType::UPGRADE))
+        return;
+
+    const String upgradeIn = hdr_in.getList(Http::HdrType::UPGRADE);
+    String upgradeOut;
+    const char *pos = nullptr;
+    const char *item;
```

Comment:
In the latest patch I removed the initialisation of ilen bellow as item for symmetry sake.

My opinion is that the best is to not initialise members when there is not any need. Initialisation of pos is required because it is used by strListGetItem.

Reply:
addressed at d55c31a

Reply:
IMO, in general, it is best to initialize variables because uninitialized variables, especially uninitialized pointers, are a common source of bugs. While any specific code may not require initialization today, many seemingly trivial modifications, often including addition of debugging, can easily add that requirement by accident.

Our inability to _meaningfully_ initialize this pointer stems from bad API. However, fixing that API is probably outside this issue scope.

Reply:
The item and ilen has meaning only inside while loop where they had set. The code is clear. IT will be huge mistake to use them outside this loop, however anyone can do it.

Not only the pointers but even integers and any variable can cause bugs so for this reason (not for symmetry) maybe it is good policy to always initialise variables.

Reply:
addressed at a2925dbf